### DISCUSSION TITLE
Questions about Burgers vectors for screw dislocations in HCP systems

### DISCUSSION THREAD

**Discussion by YJoyce** [mtex-toolbox/mtex#2016] [Ask Anything] [answered]
Hi everyone!
I am trying to calculate the GND density in hcp system. Here is a question about the calculate screw dislocation in dislocationSystem.

For the definition of SlipSystem, I choose the slipSystem.basal, slipSystem.prismaticA, slipSystem.pyramidalCA, slipSystem.pyramidal2CA, and also consider the symmertry, so there are 24 edge dislocations with the dislocationSystem function. This agrees with my idea. However, there is a discrepancy between my idea and the 9 screw dislocations calculated with dislocationSystem function.

In my understanding, the Burgers vectors for screw dislocations should be selected from the edge dislocations that are unique. But the results show some new Burgers vectors for screw dislocations which I did not expect.

I don't know what is wrong, do I need to make any changes?
![1](https://github.com/mtex-toolbox/mtex/assets/147254225/df997f07-9218-44a7-9aa6-5e129169035e)
![2](https://github.com/mtex-toolbox/mtex/assets/147254225/6a7ec62f-43d4-4b27-8d03-6548fe4dac96)

Best
Joyce

**Comment by kilir** [comment]
Hi,
what version of mtex do you use?

Here's what I get:

```
cs = crystalSymmetry('hexagonal');

sS = [slipSystem.basal(cs) ...
      slipSystem.prismaticA(cs), ...
      slipSystem.pyramidalCA(cs), ...
      slipSystem.pyramidal2CA(cs)]

sS = sS.symmetrise('antipodal');

dS = dislocationSystem(sS)


dS = dislocationSystem
 symmetry: 6/mmm, X||a*, Y||b, Z||c*
 edge dislocations : 24 x 1
 Burgers vector    line vector  energy
   [1  1 -2  0]   [1 -1  0  0]       2
   [1 -2  1  0]  [-1  0  1  0]       2
  [-2  1  1  0]   [0  1 -1  0]       2
   [2 -1 -1  0]   [0  0  0  1]       2
   [1  1 -2  0]   [0  0  0  1]       2
   [1 -2  1  0]   [0  0  0 -1]       2
   [2 -1 -1  3]  [-2 -5  7  3]       2
  [-1  2 -1 -3]   [5  2 -7  3]       2
   [2 -1 -1 -3]  [-2  7 -5 -3]       2
  [-1 -1  2  3]   [5 -7  2 -3]       2
   [2 -1 -1  3]   [2 -7  5 -3]       2
   [1  1 -2  3]   [5 -7  2  3]       2
   [2 -1 -1 -3]   [2  5 -7  3]       2
   [1 -2  1 -3]   [5  2 -7 -3]       2
   [1  1 -2  3]   [7 -5 -2 -3]       2
   [1 -2  1 -3]  [-7  2  5 -3]       2
  [-1 -1  2  3]  [-7  5  2 -3]       2
  [-1  2 -1 -3]   [7 -2 -5 -3]       2
   [2 -1 -1  3]   [0 -1  1  0]       2
   [2 -1 -1 -3]   [0  1 -1  0]       2
   [1  1 -2  3]   [1 -1  0  0]       2
   [1 -2  1 -3]   [1  0 -1  0]       2
  [-1 -1  2  3]  [-1  1  0  0]       2
  [-1  2 -1 -3]  [-1  0  1  0]       2
 
 screw dislocations: 9 x 1
 Burgers vector  energy
   [1 -2  1  0]       1
   [1 -2  1 -3]       1
  [-1  2 -1 -3]       1
  [-1 -1  2  3]       1
   [1  1 -2  0]       1
   [1  1 -2  3]       1
   [2 -1 -1  0]       1
   [2 -1 -1  3]       1
   [2 -1 -1 -3]       1
```

Cheers,
Rüdiger

**Reply by YJoyce** [reply]
Hi Rüdiger,
The version I used is mtex5.8.1.
I ran your code but the burgers vector of screw dislocation and line vector of edge dislocation are different from yours, as shown follow.
 
dS = dislocationSystem (show methods, plot)
 symmetry: 6/mmm, X||a*, Y||b, Z||c*
 edge dislocations : 24 x 1
 Burgers vector    line vector  energy
  [-1  2 -1  0]   (1  0 -1  0)       2
   [1  1 -2  0]   (1 -1  0  0)       2
   [2 -1 -1  0]   (0 -1  1  0)       2
  [-2  1  1  0]   (0  0  0 -1)       2
   [1  1 -2  0]   (0  0  0 -1)       2
  [-1  2 -1  0]   (0  0  0  1)       2
   [1 -2  1  3]   (7 -2 -5 -2)       2
   [1  1 -2 -3]  (-7  5  2 -2)       2
  [-1  2 -1  3]  (-7  2  5 -2)       2
  [-1 -1  2 -3]   (7 -5 -2 -2)       2
  [-1  2 -1  3]   (5  2 -7 -2)       2
  [-2  1  1  3]   (2  5 -7  2)       2
   [1 -2  1  3]  (-5 -2  7 -2)       2
   [2 -1 -1  3]  (-2 -5  7  2)       2
  [-1 -1  2 -3]  (-5  7 -2 -2)       2
   [2 -1 -1  3]   (2 -7  5 -2)       2
   [1  1 -2 -3]   (5 -7  2 -2)       2
  [-2  1  1  3]  (-2  7 -5 -2)       2
  [-1  2 -1  3]   (1  0 -1  0)       2
   [1 -2  1  3]  (-1  0  1  0)       2
   [1  1 -2 -3]  (-1  1  0  0)       2
  [-1 -1  2 -3]   (1 -1  0  0)       2
  [-2  1  1  3]   (0  1 -1  0)       2
   [2 -1 -1  3]   (0 -1  1  0)       2
 
 screw dislocations: 9 x 1
 Burgers vector  energy
  (-1  2 -1  0)       1
  (-1  2 -1  2)       1
   (1 -2  1  2)       1
   (1  1 -2 -2)       1
   (1  1 -2  0)       1
  (-1 -1  2 -2)       1
   (2 -1 -1  0)       1
  (-2  1  1  2)       1
   (2 -1 -1  2)       1

Best,
Joyce

**Reply by YJoyce** [reply]
I ran it with the latest version of Mtex 6.0 and it came out the same as 5.8.1

**Reply by kilir** [reply, marked as answer]
If you change the length of a and c, the indices of the normal vectors can/can change, however not the burgers vector.
I'm using the latest version from the development branch. The 6.0 is a different branch and still beta and not yet for anything but finding bugs. If you don't want to pull all from development at least try with fac91c496686fce2189b690372c1492e17bf7a76 and 2f4b6bb8cefc038913621b33d4fc6c0fd8ca1562 .

Cheers,
Rüdiger

**Reply by YJoyce** [reply]
Thanks for your reply, I updated the file and the problem is solved!
